pkg/workflow/engineimpl.go

```diff
@@ -247,21 +251,25 @@ func (e *EngineImpl) InvokeWithInputAndConfig(
 
 			// prepare logger
 			prefix := fmt.Sprintf("%s:%d", id.Host, e.invocationCounter)
-			e.mu.Unlock()
-
 			zlogger := e.logger.With().Str("ext", prefix).Logger()
 
+			localConfig := e.config
+			localNetwork := e.networkAccess
```

Comment:
I'd find it more intuitive, to have `localNetwork` never contain `e.networkAccess`, but the clone directly on assignment - that way it is always `local`. Same for config.